File: 05-hash-tables/python/hash_function_test.py
import unittest
from .hash_function import hash_function, HASH_SIZE

class TestHashFunction(unittest.TestCase):
    def Setup(self):
        pass

    def test_hash_function_returns_a_number(self):
        self.assertIsInstance(hash_function('string'), int)

    def test_hash_function_returns_a_result_less_than_hash_size(self):
        self.assertLess(hash_function('string'), HASH_SIZE)

    def test_hash_function_returns_calculated_result(self):
        self.assertEqual(hash_function('string'), 2)

File: 05-hash-tables/ruby/hash_function_test.rb
# frozen_string_literal: true

require 'minitest/autorun'
require_relative './hash_function'

class HashFunctionTest < Minitest::Test
  def test_hash_function_returns_an_integer
    assert_instance_of Integer, hash_function("string")
  end

  def test_hash_function_returns_a_result_less_than_hash_size
    assert_operator hash_function("string"), :<, HASH_SIZE
  end

  def test_hash_function_returns_calculated_result
    assert_equal 2, hash_function("string")
  end
end
